strawberry/lazy_type.py

```diff
@@ -38,7 +40,20 @@ def __class_getitem__(cls, params):
 
     def resolve_type(self) -> Type:
         module = importlib.import_module(self.module, self.package)
-
+        main_module = sys.modules.get("__main__", None)
+        if main_module:
+            # If lazy type points to the main module, use it instead of the imported
+            # module. Otherwise duplication checks during schema-conversion might fail.
+            # Refer to: https://github.com/strawberry-graphql/strawberry/issues/2397
+            if main_module.__spec__ and main_module.__spec__.name == self.module:
+                module = main_module
+            elif hasattr(main_module, "__file__") and hasattr(module, "__file__"):
+                if (
+                    main_module.__file__
+                    and module.__file__
+                    and Path(main_module.__file__).samefile(module.__file__)
+                ):
+                    module = main_module
```

Comment:
I didn't know about `samefile` :D

is this only for code that imports things from `__main__`?

Reply:
@patrick91 yes unfortunately it is an annoying edge-case where you end up with two distinct class objects that are actually from the same module. One of them will have a `__module__ = "__main__"` and the other something like `__module__` = "foo.bar". I guess this situation with `strawberry.lazy` isn't very likely, but I did run into it in my minimal repro

Reply:
ok, good to know! thanks!